src/web/app/components/comment-box/comment-table/comment-row.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { CommentModelDefaultValues, FeedbackResponseCommentModel } from './comment-table-model';
+
+/**
+ * Comment row component to be used in a comment table
+ */
+@Component({
+  selector: 'tm-comment-row',
+  templateUrl: './comment-row.component.html',
+  styleUrls: ['./comment-row.component.scss'],
+})
+export class CommentRowComponent implements OnInit {
+  @Input() isInEditModeInitialValue: boolean = false;
+  @Input() isVisibilityOptionEnabled: boolean = true;
+  @Input() isDiscardButtonEnabled: boolean = true;
+  @Input() isSaveButtonEnabled: boolean = true;
+  @Input() isReadOnlyMode: boolean = false;
+
+  @Input()
+  commentModel: FeedbackResponseCommentModel = {
+    commentId: CommentModelDefaultValues.INVALID_VALUE,
+    createdAt: CommentModelDefaultValues.INVALID_VALUE,
+    commentText: '',
+    commentGiver: '',
+    timeZone: '',
+    showCommentTo: [],
+    showGiverNameTo: [],
+  };
+
+  @Output() editCommentEvent: EventEmitter<any> = new EventEmitter();
+  @Output() deleteCommentEvent: EventEmitter<any> = new EventEmitter();
+  @Output() saveCommentEvent: EventEmitter<any> = new EventEmitter();
+  @Output() commentFormChangeEvent: EventEmitter<any> = new EventEmitter();
+
+  isInEditMode: boolean = false;
```

Comment:
`isInEditMode` is updated within the `comment-row` component. Kept them separate to avoid modifying the `@Input() isInEditModeInitialValue` from parent..